test/e2e/create_test.go

```diff
@@ -128,13 +141,25 @@ func (tc *testContext) waitForWindowsMachines(machineCount int, phase string) er
 			log.Printf("waiting for %d/%d Windows Machines", machineCount-len(machines.Items), machineCount)
 			return false, nil
 		}
+		// A phase of "" skips the phase check
+		if phase == "" {
+			return true, nil
+		}
 		for _, machine := range machines.Items {
 			if machine.Status.Phase == nil || *machine.Status.Phase != phase {
 				return false, nil
 			}
 		}
 		return true, nil
 	})
+	if phase == "" {
+		phase = "deleted"
+	}
+
+	// Log the time elapsed while waiting for creation of the Machines
+	endTime := time.Now()
+	log.Printf("%v time is required for %v Machines to reach phase %s", endTime.Sub(startTime), machines.Items, phase)
```

Comment:
You need to do `len(machine.Items)` here and replace the %v with a %d.

Reply:
Thanks for the catch @aravindhp, that was a silly one from my side! PTAL when you get a chance.